cram_pr2/cram_pr2_shopping_demo/src/plans.lisp

```diff
@@ -0,0 +1,173 @@
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -1 -0.35 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+(defparameter *pose-grasping-right* (cl-transforms-stamped:make-pose-stamped
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -2 -0.35 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+
+(defparameter *pose-near-table* (cl-transforms-stamped:make-pose-stamped
+                                  "map" 0.0
+                                  (cl-transforms:make-3d-vector -2.5 0.5 0)
+                                  (cl-transforms:axis-angle->quaternion
+                                   (cl-transforms:make-3d-vector 0 0 1) (/ pi 1))))
+
+(defparameter *looking-pose-mid* (cl-transforms-stamped:make-pose-stamped
+                                  "map" 0.0
+                                  (cl-transforms:make-3d-vector -1.5 -1.05 1.15)
+                                  (cl-transforms:make-identity-rotation)))
+
+(defparameter *pose-detecting* (cl-transforms-stamped:make-pose-stamped
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -1.5 1 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+(defun move-object (?object)
+    (let ((?grasping-arm :right)
+          (?grasp-pose *pose-grasping*)
+          (?grasp-pose-left *pose-grasping-left*)
+          (?grasp-pose-right *pose-grasping-right*)
+          (?pose-near-table *pose-near-table*)
+          (?table *table*)
+          (map->obj NIL)
+          (map->base NIL)
+          (trans NIL))
+
+      (exe:perform 
+       (desig:a motion (type moving-torso)
+                (joint-angle 0)))
+      
+      (exe:perform
+       (desig:an action (type going)
+                 (target (desig:a location (pose ?grasp-pose)))))
+      
+
+      (exe:perform
+       (desig:an action
+                 (type positioning-arm)
+                 (left-configuration park)
+                 (right-configuration park)))
+      
+
+      ;; creating the new Transform from the base_footprint infront of the shelf and the object
+      (setf map->base (cl-transforms:make-transform
+                       (cl-transforms:make-3d-vector -1.5 -0.3 0)
+                       (cl-transforms:axis-angle->quaternion
+                        (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+      
+      (setf map->obj (second (fourth (desig:desig-prop-value ?object :pose))))
+      (setf trans (cl-transforms:transform* (cl-transforms:transform-inv map->base) map->obj))
+
+      (setf ?newpose (cl-transforms-stamped:make-pose-stamped
```

Comment:
Was `?newpose` and `?newTransform` declared before? If not, include a declaration in the LET block.